src/App.js

```diff
@@ -1,12 +1,36 @@
-import './css/App.css';
+import { useState, useEffect } from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import './css/style.css';
+import axios from 'axios';
 import Navbar from './components/layout/Navbar';
 import Home from './components/pages/Home';
 import Deals from './components/pages/Deals';
 import Cart from './components/pages/Cart';
-//importing react router
-import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import ItemPage from './components/pages/ItemPage';
 
 function App() {
+  // lifting up the items state so the results can be passed to child pages
+  const [items, setItems] = useState(null);
+
+  useEffect(() => {
```

Comment:
The data should be loaded in the page components. Loading the data here will cause the data to be loaded when the user loads the app, regardless if they need it or not. We don't want to load necessary data if we can avoid it. Loading the data here will also make search and pagination much more challenging to implement in the future.

Reply:
fixed in a9af71b2e0c082692b9402817ed3a70cd11571db